internal/pkg/agent/application/upgrade/marker_access_common.go

```diff
@@ -7,10 +7,14 @@ package upgrade
 import (
 	"fmt"
 	"os"
+
+	"github.com/elastic/elastic-agent-libs/file"
 )
 
 func writeMarkerFileCommon(markerFile string, markerBytes []byte, shouldFsync bool) error {
-	f, err := os.OpenFile(markerFile, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	tmpMarkerFile := fmt.Sprintf("%s.tmp", markerFile)
```

Comment:
Only one writer at a time is possible, so we don't need to worry about collisions in this file name?

Reply:
I'll add PID and timestamp to the temporary filename just to be safe.

Reply:
Added in 09e6028a28.

Reply:
Thanks!